cestzam-core/src/main/java/be/valuya/cestzam/core/auth/emailcode/EmailCodeSessionHandler.java

```diff
@@ -0,0 +1,48 @@
+package be.valuya.cestzam.core.auth.emailcode;
+
+import be.valuya.cestzam.core.auth.CestzamAuthHandler;
+import be.valuya.cestzam.core.session.CestzamSession;
+import be.valuya.cestzam.core.session.cookie.CestzamCookies;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class EmailCodeSessionHandler implements CestzamAuthHandler {
+
+    @Override
+    public String getMethod() {
+        return EmailCodeSession.EMAIL_CODE;
+    }
+
+    @Override
+    public CestzamSession processAuthSession(CestzamSession cestzamSession) {
+        EmailCodeSession authSession = (EmailCodeSession) cestzamSession.getAuthSession();
+        EmailCodeSessionStatus status = authSession.getStatus();
+        return switch (status) {
+            case NEW_SESSSION -> handleNewSession(cestzamSession, (EmailCodeSession.New) authSession);
+            case CODE_REQUEST -> handleCodeRequestSesssion(cestzamSession, (EmailCodeSession.CodeRequest) authSession);
+            case AUTHENTICATED -> cestzamSession;
+            case EXPIRED -> cestzamSession;
+            case ERRORED -> cestzamSession;
+        };
+    }
+
+    private CestzamSession handleNewSession(CestzamSession cestzamSession, EmailCodeSession.New authSession) {
+        String username = authSession.getUsername();
+        String password = authSession.getPassword();
+        String configuredRecipientEmail = authSession.getConfiguredRecipientEmail();
+        CestzamCookies cookies = cestzamSession.getCookies();
+
+        // TODO submit username/password form & wait for mail at configuredRecipientEmail
+        EmailCodeSession.Errored updatedSession = new EmailCodeSession.Errored("Not implemented");
+        return new CestzamSession(cookies, updatedSession);
+    }
+
+    private CestzamSession handleCodeRequestSesssion(CestzamSession cestzamSession, EmailCodeSession.CodeRequest authSession) {
+        CestzamCookies cookies = cestzamSession.getCookies();
+
+        // TODO poll
+        EmailCodeSession.Errored updatedSession = new EmailCodeSession.Errored("Not implemented");
```

Comment:
the user can then poll until code is processed.

alternatively, the user should be able to provide the code in another request